README.md

```diff
@@ -1,3 +1,233 @@
+(rootless)._
+
+### Requirements for development
+
+While we run the services inside a Kubernetes cluster we recomend to install
+Go and other tools locally to ease the development process.
+
+- [Go] (1.22+)
+- GNU [Make] and [GCC]
+
+### Set up
+
+Start a local Kubernetes cluster with a local registry. For example, with k3d:
+
+```bash
+k3d cluster create preprocessing --registry-create sdps-registry
+```
+
+Or using an existing registry:
+
+```bash
+k3d cluster create preprocessing --registry-use sdps-registry
+```
+
+Make sure kubectl is available and configured to use that cluster:
+
+```bash
+kubectl config view
+```
+
+Clone this repository and move into its folder if you have not done that
+previously:
+
+```bash
+git clone git@github.com:artefactual-sdps/preprocessing-moma.git
+cd preprocessing-moma
+```
+
+Bring up the environment:
+
+```bash
+tilt up
+```
+
+While the Docker images are built/downloaded and the Kubernetes resources are
+created, hit `space` to open the Tilt UI in your browser. Check the [Tilt UI]
+documentation to know more about it.
+
+### Live updates
+
+Tilt, by default, will watch for file changes in the project folder and it will
+sync those changes, rebuild the Docker images and recreate the resources when
+necessary. However, we have _disabled_ auto-load within the Tiltfile to reduce
+the use of hardware resources. There are refresh buttons on each resource in the
+Tilt UI that allow triggering manual updates and re-executing jobs and local
+resources. You can also set the `trigger_mode` env string to `TRIGGER_MODE_AUTO`
+within your local `.tilt.env` file to override this change and enable auto mode.
+
+### Stop/start the environment
+
+Run `ctrl-c` on the terminal where `tilt up` is running and stop the cluster
+with:
+
+```bash
+k3d cluster stop preprocessing
+```
+
+To start the environment again:
+
+```bash
+k3d cluster start preprocessing
+tilt up
+```
+
+### Clear the cluster
+
+> Check the Tilt UI helpers below to just flush the existing data.
+
+To remove the resources created by Tilt in the cluster, execute:
+
+```bash
+tilt down
+```
+
+Note that it will take some time to delete the persistent volumes when you
+run `tilt down` and flushing the existing data does not delete the cluster.
+To delete the volumes immediately, you can delete the cluster.
+
+### Delete the cluster
+
+Deleting the cluster will remove all the resources immediatly, deleting
+cluster container from the host. With k3d, run:
+
+```bash
+k3d cluster delete preprocessing
+```
+
+### Tilt environment configuration
+
+A few configuration options can be changed by having a `.tilt.env` file
+located in the root of the project. Example:
+
+```text
+TRIGGER_MODE_AUTO=true
+```
+
+#### TRIGGER_MODE_AUTO
+
+Enables live updates on code changes for the preprocessing worker.
+
+### Tilt UI helpers
+
+#### Submit
+
+In the Tilt UI header there is a cloud icon/button that allows to trigger the
```

Comment:
Edit: "...a cloud icon/button that ~allows to~can trigger the preprocessing workflow."